### PR TITLE
Fix hovers and fix move unit duplicating state

### PR DESCRIPTION
**Pull request by zemiret**

### CODE REVIEW

File: communicator/room.go [hunk 1]
```diff
@@ -439,11 +439,9 @@ func (r *Room) handlePlaceUnit(client *Client, payload PlaceUnitPayload) {
 		return
 	}
 
-	moved := false
 	for i, existing := range state.UnitsPlacement {
-		if existing.UnitID == placement.UnitID {
-			moved = true
-			state.UnitsPlacement[i] = placement
```
Comment (maciekmm) [inline communicator/room.go:446]: I don't see what has changed here. The outcome should be the same (?)

File: frontend/src/components/BoardUnit.tsx [hunk 2]
```diff
@@ -20,16 +20,14 @@ const BoardUnit: React.FunctionComponent<Props> = ({ unit }: Props) => {
           onMouseEnter={() => setHover(true)}
```
Comment (maciekmm) [inline frontend/src/components/BoardUnit.tsx:20]: Do we still need this if the trigger has been set to `hover`?

File: frontend/src/components/BoardUnit.tsx [hunk 3]
```diff
@@ -11,6 +10,7 @@ interface Props {
 
 const BoardUnit: React.FunctionComponent<Props> = ({ unit }: Props) => {
   const [hover, setHover] = useState<boolean>(false);
+  const [unitId] = useState<string>(unit.id);
```
Comment (maciekmm) [inline frontend/src/components/BoardUnit.tsx:13]: Could you elaborate how these changes fix the hover being broken?

File: frontend/src/components/BoardUnit.tsx (hunk 2 above)
Comment (zemiret) [reply frontend/src/components/BoardUnit.tsx:20]: Actually, yes! It does not really matter what the trigger is, I believe it's covered by some other element.

File: frontend/src/components/BoardUnit.tsx (hunk 3 above)
Comment (zemiret) [reply frontend/src/components/BoardUnit.tsx:13]: The problem was that
1. There is a unit with id=1 on board, and so with element id `#unit-1`
2. I change units on board
3. The popover is trying to open itself but it has not yet gotten the new state (so it still thinks that it should show itself for the old id), but the DOM is already updated, so `#unit-1` is not present there

When you change this to be the state, the component will re-render in proper order. So first, it gets the new state (from redux), than it changes its internal state, and then it updates itself.

Comment (zemiret) [reply frontend/src/components/BoardUnit.tsx:13]: This gave me a clue: https://github.com/reactstrap/reactstrap/issues/773

File: communicator/room.go (hunk 1 above)
Comment (zemiret) [reply communicator/room.go:446]: Nope. There was a bug.
1. Place unit1 on the gameboard (x = 1, y = 1)
2. Place unit2 on the gameboard (x = 5, y = 5)
3. With unit2 still selected, place it on (1, 1)

Then I expect the unitsPlacement state to be
```
unit2 x=1, y=1
```

But it was:
```
unit2 x=1, y=1
unit2 x=1, y=1
```

It's easy to see that the previous code never removed anything from the gameboard. 
But in this case you had 2 units on the gameboard, and when you moved the one onto the other, the other should disappear, and be replaced.

**Pull request merged**